tests/simulation/test_cluster.hpp

```diff
@@ -0,0 +1,251 @@
+  const auto property_id_1 = properties.at("property_1");
+  const auto property_id_2 = properties.at("property_2");
+  const auto type_1 = memgraph::common::SchemaType::INT;
+  const auto type_2 = memgraph::common::SchemaType::INT;
+
+  // register new label space
+  std::vector<SchemaProperty> schema = {
+      SchemaProperty{.property_id = property_id_1, .type = type_1},
+      SchemaProperty{.property_id = property_id_2, .type = type_2},
+  };
+
+  std::optional<LabelId> label_id = sm.InitializeNewLabel(label_name, schema, replication_factor, sm.shard_map_version);
+  RC_ASSERT(label_id.has_value());
+
+  // split the shard at N split points
+  for (int64_t i = 1; i < n_splits; ++i) {
+    const auto key1 = memgraph::storage::v3::PropertyValue(i);
+    const auto key2 = memgraph::storage::v3::PropertyValue(0);
+
+    const auto split_point = {key1, key2};
+
+    const bool split_success = sm.SplitShard(sm.shard_map_version, label_id.value(), split_point);
+
+    RC_ASSERT(split_success);
+  }
+
+  return sm;
+}
+
+void ExecuteOp(msgs::ShardRequestManager<SimulatorTransport> &shard_request_manager,
+               std::set<CompoundKey> &correctness_model, CreateVertex create_vertex) {
+  const auto key1 = memgraph::storage::v3::PropertyValue(create_vertex.key);
+  const auto key2 = memgraph::storage::v3::PropertyValue(create_vertex.value);
+
+  std::vector<msgs::Value> primary_key = {msgs::Value(int64_t(create_vertex.key)),
+                                          msgs::Value(int64_t(create_vertex.value))};
+
+  if (correctness_model.contains(std::make_pair(create_vertex.key, create_vertex.value))) {
+    // TODO(tyler) remove this early-return when we have properly handled setting non-unique vertexes
+    return;
+  }
+
+  msgs::ExecutionState<msgs::CreateVerticesRequest> state;
+
+  auto label_id = shard_request_manager.NameToLabel("test_label");
+
+  msgs::NewVertex nv{.primary_key = primary_key};
+  nv.label_ids.push_back({label_id});
+
+  std::vector<msgs::NewVertex> new_vertices;
+  new_vertices.push_back(std::move(nv));
+
+  auto result = shard_request_manager.Request(state, std::move(new_vertices));
+  if (result.size() != 1) {
+    spdlog::error("did not get a result back when creating the vertex");
+    RC_ASSERT(result.size() == 1);
+  }
+
+  correctness_model.emplace(std::make_pair(create_vertex.key, create_vertex.value));
+}
+
+void ExecuteOp(msgs::ShardRequestManager<SimulatorTransport> &shard_request_manager,
+               std::set<CompoundKey> &correctness_model, ScanAll scan_all) {
+  msgs::ExecutionState<msgs::ScanVerticesRequest> request{.label = "test_label"};
+
+  auto results = shard_request_manager.Request(request);
+
+  RC_ASSERT(results.size() == correctness_model.size());
+
+  for (const auto &vertex_accessor : results) {
+    const auto properties = vertex_accessor.Properties();
+    const CompoundKey key = std::make_pair(properties[0].second.int_v, properties[1].second.int_v);
```

Comment:
I think using `VertexAccessor::Id` or adding `VertexAccessor::PrimaryKey` is better than expecting the order of the properties given.